src/plugins/es_ui_shared/public/console_lang/lib/json_xjson_translation_tools/parser.ts

```diff
@@ -0,0 +1,92 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+type StringValues = Array<{ startIndex: number; endIndex: number }>;
+
+interface ParseResult {
+  stringValues: StringValues;
+}
+
+interface ParseOptions {
+  assumeValidJson?: boolean;
+}
+
+const JSON_COLON = ':';
+const JSON_STRING_DELIMITER = '"';
+const JSON_STRING_ESCAPE = '\\';
+
+export const extractJSONStringValues = (
+  input: string,
+  { assumeValidJson = true }: ParseOptions = {}
+): ParseResult => {
+  if (!assumeValidJson) {
+    JSON.parse(input);
```

Comment:
Do you mind adding a comment here of what we expect this to do?

I am _guessing_ (but not quite sure) that we expect it to throw somehow for the consumer, to it can catch it. Wouldn't it be better to have an explicit error code for that?

```js
if (!assumeValidJson) {
  try {
    JSON.parse(input);
  } catch (e) {
    const error = new Error('The provided JSON is invalid');
    (error as any).code = 'INVALID_JSON';
    throw error;
  }
}
```

But I am still not sure what scenario we are trying to cover here.

Reply:
That would be better, and that is exactly what that code is for. At this point I think we can actually remove because the scenario I had in mind will probably not come about. Will remove this!